### DISCUSSION TITLE
Battery and Smart Meter not showing after installing 2nd Inverter

### DISCUSSION THREAD

**Discussion by neuleo** [wlcrs/huawei_solar#717] [Setting up the integration] [answered]
Hey Guys, I have a Question. 
I already set this Integration up once. It worked. Everything I connected to the inverter (5kwh) (Battery and Smart Meter) worked and showed up in HA.

Now we got a Second Inverter (7kwh) installed, and everything got reseted and the Inverter got a new IP Address. I deleted the old connection in HA and added the right IP Adress. So far everything works, execpt: The Battery and the Smart Meter doesn't show up in HA anymore. 

The 7kwh one is the Master and the 5kwh is the slave. According to the Power Figures, the Inverter with the IP Adress HA connected to, is the 5kwh, which is the Slave. 

But there is no other lwip device on my Network.
Also only the 7kwh Inverter is hooked up to the Dongle. The 5kwh is directly connected to the 7kwh inverter

How can I connect everything to home assistant?

**Comment by Roving-Ronin** [comment]
@neuleo you fail to mention the models, I assume your connecting to some L0 / L1 (i.e. single phase) inverters? If so, IMHO I would suggest you invest in a US$60 sDongle, that you plug into the Primary inverter, that will connection it all up to FusionSolar and (when you enable modbus access on the sDongle) will also allow you to connect to the Primary Inverter over your WiFi or Ethernet (assign a static IP to the dongle also). If you can connect to the Primary inverter, and it is correctly wired up via RS485 to the meter and battery, then the primary inverter will see the battery / meter and also see the downstream inverter.

You don't mention it above, that you have rewired the connections so the existing meter is now attached to the primary inverter.....? 
sDongle + Meter MUST be wired into the Primary Inverter to work.   If you have say a 5kW L0 and a 7kW L1, the L1 (or M1 if 3 phase) MUST be the primary inverter.

**Reply by neuleo** [reply]
@Roving-Ronin Thanks for your Answere. I will check that, if I get Home. Then I will provide the additional Infos, you asked for👍

**Reply by neuleo** [reply]
Hi @Roving-Ronin,

Thank you for your response and assistance. Sorry for the late Answere. Here is the additional information you requested:

1. The 5 kW inverter is a Sun2000-5KTL-M1 and the 8 kW inverter is a Sun2000-8KTL-M1.
2. Both inverters are 3-phase.
3. The Smart Meter is connected to the 8 kW inverter.
4. The RS485 connection between the 8 kW inverter and the Smart Meter is correctly established.
5. Modbus access is definitely enabled on the sDongle for the 5 kW inverter. I will check if it is also enabled for the 8 kW inverter.
6. The sDongle has a static IP address assigned.

Additionally, I initially mentioned we had a 7 kW inverter, but it is actually an 8 kW inverter. Everything is displayed correctly and functions properly in FusionSolar. The dongle is connected to the 8 kW inverter.

I will verify and ensure that Modbus access is enabled on the 8 kW inverter. Is there anything else I should check or consider?

Thank you in advance for your help!

Best regards,  
neuleo

**Reply by Roving-Ronin** [reply]
Sorry it bit confusing the way you've written it. It sounds like you might have 2 dongles installed ?

If so you want to only have one dongle, that is plugged into the 8kW (with the meter / battery also connect to the 8kW. The 5kW only has the downstream rs485 connection from the 8kW to the 5kW.

Also, as M1 inverters come with a dongle included in the box, make sure you plug the existing dongle (that was previously used to connect the 5kW to SolarFusion) is the sole dongle now used in the 8kW.

Using the old dongle with allow the 'plant' in FusionSolar to remain active and then just add the inverter(s) into this existing plant, meaning you then also keep that plants old data.

**Reply by neuleo** [reply]
@Roving-Ronin Hi,

Thank you for your assistance so far. I've verified the setup and connections as you suggested:

- The 8 kW inverter (Sun2000-8KTL-M1) is the primary inverter.
- The Smart Meter and battery are connected to the 8 kW inverter.
- The RS485 connection between the 8 kW inverter and the Smart Meter is correctly established.
- Modbus access is enabled on both inverters.
- The sDongle, with a static IP address, is connected to the 8 kW inverter. There is only one dongle in use.
- The 5 kW inverter (Sun2000-5KTL-M1) is connected downstream to the 8 kW inverter via RS485.

Despite these steps, the battery and Smart Meter still do not show up in Home Assistant (HA). Everything, including both inverters, displays correctly and functions properly in FusionSolar.

Is there anything else I should check or any further steps I need to take to ensure that the battery and Smart Meter show up in HA?

Thank you again for your help!

**Reply by Roving-Ronin** [reply, marked as answer]
@neuleo 
Have you confirmed the slave ID's assigned to each inverter? https://github.com/wlcrs/huawei_solar/wiki/Connecting-to-daisy-chained-inverters#finding-the-correct-slave-ids-to-use
and then added them in the correct order, when adding the plant in HA? I would assume you'll find that your 5kW, as it was the first installed, would have defaulted to '1' and then the 8kW being added later went to '2', so you'd be entering it as '2, 1' when setting up in HA.

**Reply by neuleo** [reply]
Bro, I can't thank you enough. You were right. I tried multiple slave ID's, but never the correct order. Thank you for your help!!
In the End it was 1,2 Slave ID. 

Have an amazing Sunday!
